dashboard/app/models/user_script.rb

```diff
@@ -36,4 +38,17 @@ def empty?
     # a user script is empty if there is no progress
     started_at.nil? && completed_at.nil? && assigned_at.nil? && last_progress_at.nil?
   end
+
+  def check_plc_task_assignment_updating
+    if check_completed? && self.script.pd
+      #Get all the task assignments for the user for script level tasks
+
+      #I'm not wild about the joinery here - we can definitely save time by adding denormalizing user_id on the
+      #enrollment_task_assignment table. The number of script completions tasks associated with a user won't be that
+      #high, so it's okay to select all of them and pick the ones we want
+      task_assignments_for_user = (Plc::UserCourseEnrollment.where(user: user).map {|x| x.plc_task_assignments.all}).flatten
+      tasks_to_update = task_assignments_for_user.select {|x| x.plc_task.type == 'Plc::ScriptCompletionTask' && x.plc_task.script_id == script.id}
+      tasks_to_update.map(&:complete_assignment)
```

Comment:
This can probably be tightened up by reducing to a single ActiveRecord-managed MySQL query rather than multiple lines of Ruby-object code. Not urgent while this feature is still WIP-mode, but I'd say it needs to be done before this is production-ready.

Not sure what the exact syntax would be (I haven't grokked every part of the whole model here), but something along the lines of `Plc::UserCourseEnrollment.joins(:plc_task, :plc_task_assignments).where(user: user, plc_task_assignments: {type: 'Plc::ScriptCompletionTask'}, plc_task: {script_id: script_id}).update(status: :completed)`.

Either way, whether an ActiveRecord-syntax join is used or the table is eventually denormalized, I'd recommend setting up a model test to fix the expected behavior of this method (to make sure the behavior applied to the model by this method remains invariant), then refactoring the implementation afterwards.